bin/cli.js

```diff
@@ -118,6 +119,19 @@ if (config.tsConfig) {
 	config.tsConfig = obj.raw;
 }
 
+if ('depth' in program) {
+	config.depth = program.depth;
+}
+
+if ('depth' in config) {
+	config.depth = Number(config.depth);
```

Comment:
I'm not sure how would this interact with the fact that `Number('')==0`. It seems `.option` in `Commander.js` is always parsed as a string. and `parseInt` was used in its [docs](https://github.com/tj/commander.js/#custom-option-processing)

Also I see more [options from StackOverflow](https://stackoverflow.com/questions/10834796/validate-that-a-string-is-a-positive-integer)

Due to lack of test on CLI, I'm not sure if there are bugs. Maybe ignore this for now.

This PR is huge. I'll need to review it much later.